gensim/models/experimental/drmm_tks.py

```diff
@@ -0,0 +1,595 @@
+from gensim.test.utils import get_tmpfile
+from gensim.models import KeyedVectors
+from collections import Counter
+from custom_losses import rank_hinge_loss
+from custom_layers import TopKLayer
+from sklearn.preprocessing import normalize
+from custom_callbacks import ValidationCallback
+from gensim import utils
+
+try:
+    import keras.backend as K
+    from keras import optimizers
+    from keras.losses import hinge
+    from keras.models import Model
+    from keras.layers import Input, Embedding, Dot, Dense, Reshape, Dropout
+    import tensorflow
+    tensorflow.set_random_seed(101010)
+    KERAS_AVAILABLE = True
+except ImportError:
+    KERAS_AVAILABLE = False
+
+random.seed(101010)
+numpy.random.seed(101010)
+logger = logging.getLogger(__name__)
+
+
+class DRMM_TKS(utils.SaveLoad):
+    """User friendly model for training on similarity learning data.
+    You only have to provide sentences in the data as a list of words.
+
+    Example Usage:
+    -------------
+    drmm_tks_model = DRMM_TKS_Model(queries, docs, labels, word_embedding_path)
+    drmm_tks_model.predict(test_queries, test_docs)
+
+    The data should have the format:
+    queries = ["When was World Wat 1 fought ?".split(),
+             "When was Gandhi born ?".split()]
+
+    docs = [
+            ["The world war was bad".split(),
+            "It was fought in 1996".split()],
+            ["Gandhi was born in the 18th century".split(),
+             "He fought for the Indian freedom movement".split(),
+             "Gandhi was assasinated".split()]
+           ]
+
+    labels = [[0, 1],
+              [1, 0, 0]]
+    """
+
+    def __init__(self, queries, docs, labels, word_embedding_path=None,
+                 text_maxlen=200, keep_full_embedding=True, normalize_embeddings=True,
+                 epochs=10, unk_handle_method='zero', validation_data=None):
+        """Initializes the model and trains it
+
+        Parameters:
+        -----------
+        queries: list of list of string words
+            The questions for the similarity learning model
+            Example:
+            queries=["When was World Wat 1 fought ?".split(),
+                     "When was Gandhi born ?".split()],
+
+        docs: list of list of list of string words
+            The candidate answers for the similarity learning model
+
+            Example:
+            docs = [
+                    ["The world war was bad".split(),
+                    "It was fought in 1996".split()],
+                    ["Gandhi was born in the 18th century".split(),
+                     "He fought for the Indian freedom movement".split(),
+                     "Gandhi was assasinated".split()]
+                   ]
+
+        labels: list of list of ints
+            Indicates when a candidate document is relevant to a query
+            1 : relevant
+            0 : irrelevant
+
+            Example:
+            labels = [[0, 1],
+                      [1, 0, 0]]
+
+        word_embedding_path: str
```

Comment:
where is KeyedVector support?

Reply:
P/S also, for parameters that not mandatory, you should add `, optional` after type (here and everywhere)

Reply:
https://github.com/aneesh-joshi/gensim/blob/07da82b968236ac063e1c08aa98ccf5080b4bb10/gensim/models/experimental/drmm_tks.py#L170

This is where I have keyed vectors

Reply:
@aneesh-joshi I mean passing kv AS input (not only filename that will be read by model)